packages/components/src/autocomplete/autocompleter-ui.js

```diff
@@ -40,6 +41,12 @@ export function getAutoCompleterUI( autocompleter ) {
 			value,
 		} );
 
+		const popoverRef = useRef();
+
+		useOnClickOutside( popoverRef, () => {
+			reset();
+		} );
```

Comment:
We're currently passing a new function every render, meaning that the `useEffect` inside `useOnClickOutside` will re-run every time.

Instead, we could either:
- pass `reset` directly: `useOnClickOutside( popoverRef, reset )`
- extract the callback as a separate variable and wrap in in`useCallback`

Reply:
Passing `reset` directly sounds good. I think I accidentally left it this way when I was doing some testing with console.log in the callback.